### ISSUE TITLE
Errors from provider-email-sendgrid are _very_ confusing

### ISSUE THREAD

**Issue by Paratron** [strapi/strapi#14291]
I found a bug in provider-email-sendgrid which caused very confusing error responses when trying to create a user.

- Node.js version: does not matter
- NPM version: does not matter
- Strapi version: 4.3.6
- Database: does not matter
- Operating system: does not matter

### Describe the bug

I was trying to register a new user through the REST API. The API response was status 400: Forbidden.
The weird thing: a database record for the new user was created! No confirmation mail was sent out.

At first, I thought it could not be the email plugin because when I hit the "Send test mail" button in the admin UI, I received a test mail just fine.

In the end, it still turned out to be the sendgrid email plugin. What I did not take into consideration was that even tough in `/config/plugins.js` I configured a `defaultFrom` address, this is eventually overridden through a default address stored with the E-Mail Address Confirmation template. There is `no-reply@strapi.io` set by default which caused sendgrid to refuse to send the mail because thats not a valid sender address for my account.

Now the problem is that in fact, sendgrid response with `400: forbidden` here, its still a bug that THIS EXACT error bubbles up until the REST endpoint of strapi ultimatively returns this to the browser. There is _nothing_ in the console of the strapi node process.

The plugin throws the error object, here: https://github.com/strapi/strapi/blob/main/packages/providers/email-sendgrid/lib/index.js#L29

However, the code upwards looks only for the `message` property of the error, which is in fact "forbidden". But there is much more information here:

```json
{
  "message": "Forbidden",
  "code": 403,
  "response": {
    "body": {
      "errors": [
        {
          "message": "The from address does not match a verified Sender Identity. Mail cannot be sent until this error is resolved. Visit https://sendgrid.com/docs/for-developers/sending-email/sender-identity/ to see the Sender Identity requirements",
          "field": "from",
          "help": null
        }
      ]
    }
  }
}
```
The sendgrid plugin should actually look into the error object coming from sendgrid and at least provide some meaningful message in the node console instead of throwing the whole object upwards to code which is not aware of the additional information inside the `response` property.

So what I think should be done (thats full up to you tough):
- Fix the Sendgrid plugin and write detailed error information into the Strapi process' console
- Catch errors during confirmation mail processing differently. Do NOT pass them to the REST interface. The REST interface should still respond with the data of the created user
- Remove `no-reply@strapi.io` as default sender address from the email templates. Just leave it empty so the configured default address from the config file can be used. This would boost UX enormous!

### Steps to reproduce the behavior

1. Install and configure the sendgrid plugin
2. Leave confirmation template with the default strapi address
3. Register a new user with enabled email confirmation
4. See error 400: Forbidden - no user data is returned, user record still created

### Expected behavior

The data of the created user should be sent in the REST api response, even tough no confirmation mail could be sent.
There should be meaningful error messages in the Strapi process logs.

**Comment by vikasvmads** [comment]
Hi, If the issue is open, I want to work on it. :)

**Comment by bayramorhan** [comment]
Up up +++

I am having this major issue when trying to send confirmation e-mail on register with SendGrid

`"@strapi/provider-email-sendgrid": "^4.6.1",`

 ```
email: {
      config: {
        provider: 'sendgrid',
        providerOptions: {
          apiKey: env('SENDGRID_API_KEY'),
        },
        settings: {
          defaultFrom: 'my verified sendgrid email',
          defaultReplyTo: 'my verified sendgrid email',
          testAddress: '*******',
        },
      },
    }
```

But strapi using **Administration Panel <no-reply@strapi.io>** as from email instead my verified sendgrid email in **email.settings.defaulfForm** at plugins.js and it returns 400 Error but still creates the user.

**Comment by Paratron** [comment]
You need to set your from mail inside the strapi admin panel settings as a workaround.

**Comment by bayramorhan** [comment]
> You need to set your from mail inside the strapi admin panel settings as a workaround.

Which version you are using? Because there is no such setting in the admin panel.

**Comment by Paratron** [comment]
Granted, UX wise its super strange but you need to go to the Email Templates section. Each template has its own email send address for some reason - and that defaults to no-reply@strapi.io

![grafik](https://user-images.githubusercontent.com/902599/220398620-39b40df9-9d61-4f7f-865f-22ccc0806e79.png)

**Comment by jj-theInvincible** [comment]
> Up up +++
> 
> I am having this major issue when trying to send confirmation e-mail on register with SendGrid
> 
> `"@strapi/provider-email-sendgrid": "^4.6.1",`
> 
> ```
> email: {
>      config: {
>        provider: 'sendgrid',
>        providerOptions: {
>          apiKey: env('SENDGRID_API_KEY'),
>        },
>        settings: {
>          defaultFrom: 'my verified sendgrid email',
>          defaultReplyTo: 'my verified sendgrid email',
>          testAddress: '*******',
>        },
>      },
>    }
> ```
> 
> But strapi using **Administration Panel [no-reply@strapi.io](mailto:no-reply@strapi.io)** as from email instead my verified sendgrid email in **email.settings.defaulfFrom** at plugins.js and it returns 400 Error but still creates the user.
> 
> It works if I change this ` from: from || settings.defaultFrom,`
> 
> to ` from: settings.defaultFrom,`
> 
> in node_modules/@strapi/provider-email-sendgrid/lib/index.js
> 
> Using this method with Nuxt 3
> 
> "@nuxtjs/strapi": "^1.7.1",
> 
> ```
> // Strapi Register Method
> const { register } = useStrapiAuth();
> ```

Same issue, it just doesn't seem to work.
Here's what I did:
- Install @strapi/provider-email-sendgrid package using
```
npm install @strapi/provider-email-sendgrid --save

```
- Update my .config/plugins.js file
```

module.exports = ({ env }) => ({
  // ...
  email: {
    config: {
      provider: 'sendgrid', // For community providers pass the full package name (e.g. provider: 'strapi-provider-email-mandrill')
      providerOptions: {
        apiKey: env('SENDGRID_API_KEY'),
      },
      settings: {
        defaultFrom: 'verifiedmail@myorg.com',
        defaultReplyTo: 'verifiedmail@myorg.com',
        testAddress: 'verifiedmail@myorg.com',
      },
    },
  },
  // ...
});
```
But after following these steps, here's the weird thing:
- It still doesn't show my email settings in configuration 
![image](https://user-images.githubusercontent.com/23653713/228090275-5488513b-f227-4eb0-99ef-0acfe20824ea.png)
- When I search for providers, it shows that sendgrid is installed
![image](https://user-images.githubusercontent.com/23653713/228090388-1f80da05-0af4-419b-8d1f-e079e443bd0a.png)

**Comment by peterbrowse** [comment]
Bump on this.

I had no issues getting the email configured or sending the test email...

However, when I'm using the 'forgot-password' endpoint of /api/auth/forgot-password, I get an error passed from sendgrid of 'forbidden' with no other details.

Sort of impossible to work out what the error is without it being handled properly.

**Comment by AaronRobinson** [comment]
@peterbrowse did you change the shipper email address to an account on your own domain? It defaults to no-reply@strapi.io and this gets rejected as a 403 but with no further detail so it's impossible to know what the issue is. Really really poor error handling.

**Comment by peterbrowse** [comment]
After a few hours, yes, I had assumed it would take the value in the .env file I’d set for the test email, but it did notttt. All good.On 16 Nov 2024, at 18:32, Aaron ***@***.***> wrote:﻿
@peterbrowse did you change the shipper email address to an account on your own domain? It defaults to ***@***.*** and this gets rejected as a 403 but with no further detail so it's impossible to know what the issue is. Really really poor error handling.

**Comment by spyderdsn** [comment]
I agree. Got stuck on the same. After reading all the threads here my issue is still present.

- Default sender email is set to no-reply@myapp.com
- Default response email is set to no-reply@myapp.com
- Both Email templates are set to no-reply@myapp.com under the shipper email
- My IP added to SendGrid

```js
email: {
     config: {
       provider: 'sendgrid',
       providerOptions: {
         apiKey: env('SENDGRID_API_KEY'),
       },
       settings: {
         defaultFrom: 'no-reply@myapp.com',
         defaultReplyTo: 'no-reply@myapp.com'
       },
     },
   }
```

My email is not verified in SendGrid, but this is not important now. The domain is dummy at this stage. What else? 

All I get is a Forbidden error, running from my localhost...

**Comment by peterbrowse** [comment]
For sendgrid to work correctly, you’d of had to have added the domain verification for ‘myapp.com’, otherwise it won’t have the correct permissions to send and therefore throw an error… it just can’t report through.On 17 Dec 2024, at 23:51, spyderdsn ***@***.***> wrote:﻿
I agree. Got stuck on the same. After reading all the threads here my issue is still present.

Default sender email is set to ***@***.***
Default response email is set to ***@***.***
Both Email templates are set to ***@***.*** under the shipper email
My IP added to SendGrid

email: {
     config: {
       provider: 'sendgrid',
       providerOptions: {
         apiKey: env('SENDGRID_API_KEY'),
       },
       settings: {
         defaultFrom: ***@***.***',
         defaultReplyTo: ***@***.***'
       },
     },
   }
My email is not verified in SendGrid, but this is not important now. The domain is dummy at this stage. What else?
All I get is a Forbidden error, running from my localhost...
